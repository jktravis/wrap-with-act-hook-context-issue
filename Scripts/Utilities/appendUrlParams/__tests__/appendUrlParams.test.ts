import appendUrlParams from "../appendUrlParams";

describe("appendUrlParams", () => {
  describe("when url contains a question mark", () => {
    it("should append the params with an ampersand", () => {
      const url = "http://foo.bar?param1=value1";
      const params = {
        param2: "value2",
        param3: 3,
      };
      expect(appendUrlParams(params, url)).toEqual("http://foo.bar?param1=value1&param2=value2&param3=3");
    });
  });

  describe("when url does _not_ contain a question mark", () => {
    it("should append the params with a question mark", () => {
      const url = "http://foo.bar";
      const params = {
        param2: "value2",
        param3: 3,
      };
      expect(appendUrlParams(params, url)).toEqual("http://foo.bar?param2=value2&param3=3");
    });
  });

  describe("when params is a string instead of an object", () => {
    describe("and url does contain a question mark", () => {
      it("should append the string", () => {
        const url = "http://foo.bar?param1=value1";
        const params = "param2=value2&param3=3";
        expect(appendUrlParams(params, url)).toEqual("http://foo.bar?param1=value1&param2=value2&param3=3");
      });
    });
    describe("and url does not contain a question mark", () => {
      it("should append the string", () => {
        const url = "http://foo.bar";
        const params = "param2=value2&param3=3";
        expect(appendUrlParams(params, url)).toEqual("http://foo.bar?param2=value2&param3=3");
      });
    });
  });

  describe("when params is any other type", () => {
    describe("and url does contain a question mark", () => {
      it("should return the `toString()` value of the param and log an error to the console", () => {
        const consoleSpy = jest.spyOn(global.console, "error").mockImplementation(() => {});
        const url = "http://foo.bar?param1=value1";
        const params = ["param2", "value2"];
        expect(appendUrlParams(params, url)).toEqual("http://foo.bar?param1=value1&param2,value2");
        // @ts-ignore
        expect(consoleSpy.mock.calls).toMatchSnapshot();
        consoleSpy.mockRestore();
      });
    });
  });
});
