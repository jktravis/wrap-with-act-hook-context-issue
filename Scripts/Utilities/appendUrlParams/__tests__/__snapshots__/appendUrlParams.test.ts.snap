// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appendUrlParams when params is any other type and url does contain a question mark should return the \`toString()\` value of the param and log an error to the console 1`] = `
Array [
  Array [
    "appendUrlParams only works with objects and strings.",
  ],
  Array [
    " Please use \`appendUrlParamsWith\` to create an alternative implementation",
  ],
  Array [
    "called with: ",
    Array [
      "param2",
      "value2",
    ],
  ],
]
`;
